Add new benchmark: Spanish bench

zxcvuser opened the pull request:
SpanishBench is a benchmark for tasks in Spanish that cover several evaluation areas. The datasets consist of professional translations of relevant English datasets and newly created datasets in Spanish. The README.md contains detailed information on all the tasks included in the benchmark.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2157) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2157) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2157) it.</sub>

lintangsutawika commented on lm_eval/tasks/spanish_bench/escola.yaml:
```diff
@@ -0,0 +1,14 @@
+dataset_path: nbel/EsCoLA
+output_type: multiple_choice
+training_split: train
+validation_split: validation
```
Looks like the dataset does not have a `validation` split.
https://huggingface.co/datasets/nbel/EsCoLA

zxcvuser replied:
Thank you for bringing this to our attention. We've already contacted the dataset authors to fix this as soon as possible.

zxcvuser replied:
Since the EsCoLA dataset authors cannot make the changes for a few weeks, we've decided to temporarily remove the task from the benchmark. We have made all the necessary updates. Thank you!

lintangsutawika commented:
@zxcvuser need little help for running `pre-commit run --all-files` and should be good.

baberabb commented:
Getting a evaluate/blue related error (not due to this PR):
```
AttributeError: 'DownloadConfig' object has no attribute 'use_auth_token'
```
Will fix the underlying cause before merging!

zxcvuser commented:
These are the changes done:
- Added the benchmark info in lm_eval/tasks/README.md
- Replaced "-" by "_" in the create_files script in flores_es and added weight_by_size: false
- Run linters
- Remove grouping in paws_es, xnli, and mgsm tasks (they were pointing to pre-existing benchmarks) and renamed the tasks with the suffix "_spanish_bench", both in each of the task yamls and in the spanish_bench.yaml
- Changed "group" to "tag" in phrases_es

With these, it should all be fine now. Thank you!
